packages/cli/src/LoadNodesAndCredentials.ts

```diff
@@ -330,16 +354,35 @@ export class LoadNodesAndCredentialsClass implements INodesAndCredentials {
 				const iconUrl = `icons/${typeName}/${type.name}${path.extname(icon)}`;
 				delete type.icon;
 				type.iconUrl = iconUrl;
-				const source = path.join(dir, icon);
-				const destination = path.join(GENERATED_STATIC_DIR, iconUrl);
-				return mkdir(path.dirname(destination), { recursive: true }).then(async () =>
-					copyFile(source, destination),
-				);
+				if (!isReload) {
+					const source = path.join(dir, icon);
+					const destination = path.join(GENERATED_STATIC_DIR, iconUrl);
+					return mkdir(path.dirname(destination), { recursive: true }).then(async () =>
+						copyFile(source, destination),
+					);
+				}
+				return;
 			}),
 		);
 
 		await Promise.all(iconPromises);
 
+		// Unload everything that's should be lazy loaded so it'll get
+		// reimported later on
+		if (isReload && loader instanceof PackageDirectoryLoader) {
+			for (const type in loader.known.nodes) {
+				if (type in this.loaded.nodes) {
+					delete this.loaded.nodes[type];
```

Comment:
should we also not remove all relevant entries in `require.cache`?

Reply:
That's happening in `ReloadNodesAndCredentials.ts`. It could potentially be moved here though. I think I did it there because I wasn't sure how I'd be actually reloading the modules.